diagnostics-app/src/app/telemetry/telemetry-dashboard/telemetry-dashboard.component.ts

```diff
@@ -1,22 +1,68 @@
-// Copyright 2022 The Chromium Authors. All rights reserved.
+// Copyright 2021-2023 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
 /**
- * @fileoverview Defines the Dashboard Component.
- * This is the root layout component which holds the dashboard.
- * Imported by dashboard.module.ts
+ * @fileoverview Defines the Dashboard Component in Telemetry.
+ * This is the root layout component which holds the telemetry dashboard.
+ * Imported by telemetry.module.ts
  */
 
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { DpslTypes } from '@common/dpsl';
+import { TelemetryService } from 'src/app/core/services/telemetry.service';
+import { TelemetryInfoType } from '@common/message';
+import { refreshIntervals } from '../../core/config/data-refresh-intervals';
 
 @Component({
   selector: 'app-telemetry-dashboard',
   templateUrl: './telemetry-dashboard.component.html',
   styleUrls: ['./telemetry-dashboard.component.css'],
 })
-export class TelemetryDashboardComponent implements OnInit {
-  constructor() {}
+export class TelemetryDashboardComponent implements OnInit, OnDestroy {
+  private _displayTypes: TelemetryInfoType[] = [
+    TelemetryInfoType.BATTERY,
+    TelemetryInfoType.BLOCK_DEVICE,
+    TelemetryInfoType.CPU,
+    TelemetryInfoType.MEMORY,
+    TelemetryInfoType.STATEFUL_PARTITION,
+    TelemetryInfoType.VPD,
+  ];
+  private _intervalIds: number[] = [];
+  private _data = new Map<TelemetryInfoType, DpslTypes>([]);
 
-  ngOnInit(): void {}
+  get displayTypes() {
+    return this._displayTypes;
+  }
+
+  get data() {
+    return this._data;
+  }
+
+  private _updateData(displayType: TelemetryInfoType) {
+    this.telemetryService.fetchInfo(displayType).then((value: DpslTypes) => {
+      this._data.set(displayType, value);
+    });
+  }
+
+  constructor(private telemetryService: TelemetryService) {
+    for (let displayType of this._displayTypes) {
+      this._updateData(displayType);
+    }
+  }
+
+  ngOnInit(): void {
+    for (let displayType of this._displayTypes) {
+      let id = window.setInterval(() => {
+        this._updateData(displayType);
+      }, refreshIntervals.get(displayType));
```

Comment:
Check if  `displayType` is in `refreshIntervals`.

Reply:
Done.